template/src/navigation/NavigationService.tsx

```diff
@@ -1,21 +1,27 @@
-import * as React from 'react'
 import {StackActions} from '@react-navigation/native'
+import {createNavigationContainerRef} from '@react-navigation/native'
 import {INavigationParams} from '../constants/interface/navigation/NavigationInterface'
+import {AppStackParamList} from './StackNavigation'
+import {screenMatch} from './ScreenService'
 
-export const navigationRef: React.RefObject<any> = React.createRef()
+export const navigationRef = createNavigationContainerRef<AppStackParamList>()
 
 export function navigate(name: string, params: INavigationParams): void {
-  navigationRef.current?.navigate(name, params)
+  if (navigationRef.isReady()) {
+    navigationRef.navigate(screenMatch(name), params)
+  }
 }
 
-export const checkRouteOrigin = () => {
-  return navigationRef.current.getRootState().routeNames[0]
-}
+export const checkRouteOrigin = () => navigationRef.getRootState().routeNames[0]
 
 export function navigationPop() {
-  navigationRef.current?.dispatch(StackActions.pop(1))
+  if (navigationRef.isReady()) {
+    navigationRef.dispatch(StackActions.pop(1))
```

Comment:
Because you are updating this function, can we make it able to pass number of times we want to pop, with default value `1`, like 
```
export function navigationPop(time = 1) {
...
navigationRef.dispatch(StackActions.pop(time))
...
```

Reply:
updated